script/package

```diff
@@ -1,166 +1,55 @@
-    File.expand_path File.join(current, "..", paths)
-  end
-
-  def glob_dir(path)
-    Dir[path].select { |d| File.directory?(d) }
-  end
-
-  def parse_version!
-    content = File.read root_path("commands", "version.go")
-    match = /var Version = "(.+)"/.match content
-    raise "Fail to parse Hub version" unless match
-
-    match[1]
-  end
-
-  def install_gox!
-    puts "Installing github.com/mitchellh/gox"
-    result = system "go get github.com/mitchellh/gox"
-    raise "Fail to install gox" unless result
-  end
-
-  def build_toolchain!
-    puts "Building Go toolchain"
-    result = system "gox -build-toolchain -os=#{OS.type}"
-    raise "Fail to build Go toolchain" unless result
-  end
-
-  def run_tests!
-    puts "Running Hub tests"
-
-    bootstrap_script = root_path("script", "bootstrap")
-    exec!(bootstrap_script)
-
-    test_script = root_path("script", "test")
-    exec!(test_script)
-  end
-
-  def build_hub!
-    puts "Building for #{OS.friendly_name}"
-    release_version = `script/version`.strip
-    output = root_path("target", "{{.Dir}}-#{OS.friendly_name}-{{.Arch}}-#{version}", "{{.Dir}}")
-    # gox doesn't build for 64 bit and 32 bit on 64 bit Windows
-    # specifying osarch for Windows
-    # see https://github.com/mitchellh/gox/issues/19#issuecomment-68117016
-    osarch = OS.windows? ? "windows/#{OS.windows_64? ? "amd64" : "386"}" : ""
-    cmd = "gox -os=#{OS.type} -output=#{output} -ldflags \"-X github.com/github/hub/commands.Version #{release_version}\""
-    cmd += " -osarch=#{osarch}" unless osarch.empty?
-    exec!(cmd)
-  end
-
-  def cp_assets
-    path = root_path("target", "*#{OS.friendly_name}*")
-    glob_dir(path).each do |dir|
-      puts "Copying assets to #{dir}"
-      ["README.md", "LICENSE", "etc/", "man/"].each do |f|
-        cp_r f, File.join(dir, f)
-      end
-    end
-  end
-
-  def tar_gzip
-    path = root_path("target", "*#{OS.friendly_name}*")
-    glob_dir(path).each do |dir|
-      puts "Archiving #{dir}"
-      Dir.chdir(root_path("target")) do
-        exec!("tar -zcf #{File.basename(dir)}.tar.gz #{File.basename(dir)}")
-      end
-    end
-  end
-end
-
-Packer.pack!
+# Packages the project as a release asset and prints the archive's filename.
+set -e
+
+os="${1?}"
+arch="${2?}"
+version="${3?}"
+
+release="hub-${os}-${arch}-${version}"
+
+case "$os" in
+  darwin | freebsd | linux | windows ) ;;
+  * ) echo "unsupported OS: $os" >&2; exit 1 ;;
+esac
+
+case "$arch" in
+  386 | amd64 ) ;;
+  * ) echo "unsupported arch: $arch" >&2; exit 1 ;;
+esac
+
+export GIMME_OS="$os"
+export GIMME_ARCH="$arch"
+eval "$(gimme "$TRAVIS_GO_VERSION")" >/dev/null
+
+tmpdir="tmp/${release}"
+rm -rf "$tmpdir"
+
+exename="${tmpdir}/bin/hub"
+[ "$os" != "windows" ] || exename="${exename}.exe"
+mkdir -p "${exename%/*}"
+script/build -o "$exename"
+
+if [ "$os" = "windows" ]; then
+  cp README.md "${tmpdir}/README.txt"
+  cp LICENSE "${tmpdir}/LICENSE.txt"
+  cp man/hub.1.html "${tmpdir}/hub.html"
```

Comment:
For Windows, package docs as `.txt` so they're easily opened by Notepad. Also, copy the man page over as HTML representation since Windows doesn't have `man`.